test/linalg/generic.jl

```diff
@@ -65,26 +65,34 @@ let X = [3  9   5;
     @test_throws ArgumentError diff(X,-1)
 end
 
+# test linrange
+# make sure unequal input arrays throw an error
+x = [2; 5; 6]
+y = [3; 7; 10; 10]
+@test_throws DimensionMismatch linreg(x, y)
+x = [2 5 6]
+y = [3; 7; 10]
+@test_throws MethodError linreg(x, y) 
+
 # check (UnitRange, Array)
 x = 1:12
 y = [5.5; 6.3; 7.6; 8.8; 10.9; 11.79; 13.48; 15.02; 17.77; 20.81; 22.0; 22.99]
 @test_approx_eq linreg(x,y) [2.5559090909090867, 1.6960139860139862]
 @test_approx_eq linreg(sub(x,1:6),sub(y,1:6)) [3.8366666666666642,1.3271428571428574]
 
-##The following tests will not require collect once issue #16520 is fixed
```

Comment:
The issue is still a problem, but since this code no longer uses the `\` operator it is not effected and the tests work as intended, hence the removal of the `collect` hacks.